pkg/sentry/server/ca/ca.go

```diff
@@ -128,42 +128,61 @@ func New(ctx context.Context, conf config.Config) (Signer, error) {
 			return nil, fmt.Errorf("failed to generate ECDSA key for X.509 certificates: %w", err)
 		}
 
+		generatedBundle, err := ca_bundle.Generate(
+			ca_bundle.GenerateOptions{
+				X509RootKey:        x509RootKey,
+				TrustDomain:        conf.TrustDomain,
+				AllowedClockSkew:   conf.AllowedClockSkew,
+				OverrideCATTL:      nil,
+				MissingCredentials: ca_bundle.MissingCredentials{X509: true},
+			})
+		if err != nil {
+			return nil, fmt.Errorf("failed to generate CA bundle: %w", err)
+		}
+
+		bundle.X509 = generatedBundle.X509
+
+		if err := castore.store(ctx, bundle); err != nil {
+			return nil, fmt.Errorf("failed to store CA bundle: %w", err)
+		}
+
+		log.Info("Self-signed certs generated and persisted successfully")
+		monitoring.IssuerCertChanged()
+	} else {
+		log.Info("Root and issuer certs found: using credentials from store")
+	}
+	monitoring.IssuerCertExpiry(bundle.X509.IssChain[0].NotAfter)
+
+	if conf.JWT.Enabled && missingCreds.JWT {
+		log.Info("JWT not found: generating self signed")
+
 		// Generate a key for JWT signing
 		jwtRootKey, err := rsa.GenerateKey(rand.Reader, 2048)
 		if err != nil {
 			return nil, fmt.Errorf("failed to generate RSA key for JWT signing: %w", err)
 		}
 		conf.JWT.SigningAlgorithm = jwa.RS256.String()
 
-		generatedBundle, err := ca_bundle.Generate(ca_bundle.GenerateOptions{
-			X509RootKey:        x509RootKey,
-			JWTRootKey:         jwtRootKey,
-			TrustDomain:        conf.TrustDomain,
-			AllowedClockSkew:   conf.AllowedClockSkew,
-			OverrideCATTL:      nil,
-			MissingCredentials: missingCreds,
-		})
+		generatedBundle, err := ca_bundle.Generate(
```

Comment:
@lrascao Does this second `ca_bundle.Generate` generate multiple root keys unnecessarily ?

Reply:
Nvm, I see you are setting MissingCredentials: ca_bundle.MissingCredentials{JWT: true},